compiler/main/checks.cpp

```diff
@@ -191,6 +243,35 @@ void check_cullOverReferences()
   for_alive_in_Vec(ContextCallExpr, cc, gContextCallExprs) {
     INT_FATAL("ContextCallExpr should no longer be in AST");
   }
+
+  // for all CallExprs, if we call a promotion wrapper that is marked no promotion, warn
```

Comment:
`resolve` is what adds the promotion wrappers. But in these cases, it inserts a `ContextCallExpr` for the three variants of `this`. That does not get resolved to a single `CallExpr` until the `cullOverReferences` pass. 

This check could go later, but I think if it went after `lowerIterators` it would be harder to find the promoted functions and give good warnings (with correct line numbers)

Reply:
Ok thanks for letting me know. You may want to consider adding a comment explaining this placement incase we ever want to reorder things.

I think having it here rather than any later makes sense (the earlier you do the check the faster users get the warning and the less likely there'll be some intervening transform that mangles AST in unexpected ways).